core/code/layerchooser.js

```diff
@@ -133,6 +133,47 @@ var LayerChooser = L.Control.Layers.extend({
     return this;
   },
 
+  _onLongClick: function (idx, originalEvent) {
+    var el = this._layers[idx];
+    var defaultPrevented;
+
+    // @section Extra events
+    // Fired when the overlay's label is long-clicked in the layers control.
+    // Besides [standard](https://leafletjs.com/reference.html#layerscontrolevent) properties
+    // contains also following properties:
+
+    // @miniclass LayersControlEventExt
+    // @inherits LayersControlEvent
+    // @property control: L.Control
+    // layerChooser instance.
+    // @property idx: Number
+    // Internal index of overlay.
+    // @property originalEvent: DOMEvent
+    // The original mouse/jQuery event that triggered this Leaflet event.
+    // @property preventDefault: Function
+    // Method to prevent default action of event (like overlays toggling), otherwise handled by layerChooser.
+    var obj = {
+      control: this,
+      idx: idx,
+      originalEvent: originalEvent || {type: 'taphold'},
+      preventDefault: function () {
+        defaultPrevented = true;
+        this.defaultPrevented = true;
+      }
+    };
+
+    // @event longclick: LayersControlEventExt
+    // Fired on layer
+    el.layer.fire('longclick', obj);
+
+    // @event overlaylongclick: LayersControlEventExt
+    // Fired on map
+    this._map.fire('overlaylongclick', L.extend(obj, el));
```

Comment:
`L.extend(obj, el)` is this common pattern in Leaflet ?

Reply:
Yes.
Do you have some concerns regarding our case?

Reply:
no, it looked weird to extend event data with a layer
I expect `L.extend(obj.target/progabateTarget/sourceTarget, el)` instead
following the https://leafletjs.com/reference-1.7.1.html#event description

Reply:
`target` is set by leaflet: https://github.com/Leaflet/Leaflet/blob/0f904a515879fcd08f69b7f51799ee7f18f23fd8/src/core/Events.js#L179

> no, it looks weird to extend event data with a layer

`el` is not layer, and we do basically the same as `L.Control.Layers`: https://github.com/Leaflet/Leaflet/blob/0f904a515879fcd08f69b7f51799ee7f18f23fd8/src/control/Control.Layers.js#L303